packages/source-iottwinmaker/src/property-value/provider.ts

```diff
@@ -0,0 +1,76 @@
+import { DataRequest, Provider, ProviderObserver, DataBase, DataStreamBase, Primitive } from '@iot-app-kit/core';
+import { QueryClient, QueryObserver } from '@tanstack/react-query';
+
+import { IoTTwinMakerClient } from '@aws-sdk/client-iottwinmaker';
+import { getPropertyValueByEntity } from './client/getPropertyValueByEntity';
+import { TwinMakerStaticDataQuery } from './types';
+
+/**
+ * Provider for TwinMaker time series data
+ */
+export class TwinMakerPropertyValueDataProvider implements Provider<DataBase[]> {
+  public tmClient: IoTTwinMakerClient;
```

Comment:
Constructor takes tmClient and queryClient. Maybe we can add input validations to ensure these parameters are provided and have expected type?

Reply:
they are required parameters with correct types, so the input validation is already there